trilitech/jstz
# Fetch persistent logs

# Context

[Fetch logs endpoints](https://app.asana.com/0/1205770721173531/1206350698277900/f) 

This is the third step of the persistent logs feature as shown below: 
 
1. add request logs 
Before we store sf logs in the DB, we need to add logs indicating the start and end of smart function request. #325 

2. define a sqlite schma, add a functionality to flush the logs to DB  #334

3. Store the logs from kernel.log to the DB, add new endpoints to jstz node to enable fetching logs 



# Description

This pr adds two endpoints for fetching logs:
1.  logs/<sf_address>/persistent/requests/<request_id>` for the logs given s
2. logs/<sf_address>/persistent/requests` for paginated list of logs (optionally provide offset & limit)


# Manually testing the PR
Make sure `sqlite3` is installed.

Deploy and run the examples/logs.js smart functions a few times,  
Check if the logs are stored correctly by running:
`curl http://localhost:8933/logs/<sf_address>/persistent/requests/<request_id>`
`curl http://localhost:8933/logs/<sf_address>/persistent/requests\?offset\=3\&limit\=3`

Alternatively you can directly fetch from the db locally: 

https://github.com/trilitech/jstz/assets/128799322/59a981c5-5b35-4db4-a137-32cdf95a754d
